### ISSUE TITLE
Android: super buggy animation / iOS no borderRadius animation

### ISSUE THREAD

**Issue by Hirbod** [IjzerenHein/react-native-shared-element#14]
Hi,

using the react-navigation-shared-element to transform an image with borderRadius (using resizeMode cover) to the new screen using resizeMode contain (without borders), I get a super buggy animation on android.

Is flickering, blinking, sizing to fullscreen and back.

Some Facts:
- EXPO SDK 36 (your lib is at version 0.5.1, which I can't upgrade unless Expo releases a new update)
- I saw some commits addressing borderRadius issues on iOS? Maybe you did already fix my second issue)
- Tested on real device, standalone, with and without debug mode

I mad a simple video as gif.
![ezgif com-optimize](https://user-images.githubusercontent.com/504909/71526318-92d89000-28d6-11ea-8a3e-15ce4bf5b3cb.gif)

It's working nearly perfect on iOS (but the borderRadius animation is not working. Its switching hard)

```js
function springyFadeIn() {
    const transitionSpec = {
        timing: Animated.spring,
        speed: 22,
        bounciness: 1,
        useNativeDriver: true,
    };

    return {
        transitionSpec,
        screenInterpolator: ({position, scene}) => {
            const {index} = scene;

            const opacity = position.interpolate({
                inputRange: [index - 1, index],
                outputRange: [0, 1],
            });

            return {opacity};
        },
    };
}
```

```js
const userStack = createSharedElementStackNavigator(
    createStackNavigator, {
        User: {screen: User},
        UserImage: {screen: UserImage},
    },
    {
        ...StackNavigatorOptions,
        transitionConfig: () => springyFadeIn(),
    },
);
```

My Image Screen Component

```js
import * as React from 'react';
import {View, StyleSheet, Text, Image} from 'react-native';
import {SharedElement} from 'react-navigation-shared-element';
import {Body, Button, Container, Header, Icon, Left, Right, Spinner, Title} from "native-base";

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: "center",
        backgroundColor: "transparent"
    },
    image: {
        width: '100%',
        height: '100%',
        borderRadius: 0
    },
});

const UserImage = ({navigation}) => {
    const { image } = navigation.state.params;
    console.log(image);

    return (
        <Container>
            <Header
                style={{color: "#ffffff", backgroundColor: "#b7cc23"}}
                iosBarStyle="light-content"
                transparent
                androidStatusBarColor="#b7cc23"
            >
                <Left>
                    <Button transparent onPress={() => navigation.pop()}>
                        <Icon style={{color: "white"}} name='arrow-back'/>
                    </Button>
                </Left>
                <Body style={{flex: 3}}><Title>Text</Title></Body>
                <Right />
            </Header>
            <View style={styles.container}>
                <SharedElement id="image" style={StyleSheet.absoluteFill}>
                    <Image
                        style={styles.image}
                        resizeMode="contain"
                        source={{ uri: image}}
                    />
                </SharedElement>
            </View>
        </Container>
    );
};


// Add the `sharedElements` function to the component, which
// should return a list of shared-elements to transition.
// The `sharedElements` function is called whenever you navigate
// to or from this screen. You can use the provided navigation
// states or trigger or disable animations.
UserImage.sharedElements = (navigation, otherNavigation, showing) => [
    {id: 'image', resize: "clip"},
];

export default UserImage;
```

**Comment by Hirbod** [comment]
And this is how it looks like on iOS (pretty solid, just the borderRadius issue is annoying)

![RPReplay_Final1577464044](https://user-images.githubusercontent.com/504909/71526564-cff15200-28d7-11ea-9516-e8ff44f86610.gif)

**Comment by Hirbod** [comment]
@IjzerenHein  Ok, I was able to "fix" this on android. Using android navigator with "animation.spring" is the issue.

``` js 
const transitionSpec = {
        timing: Animated.spring,
        speed: 35,
        bounciness: 0,
        useNativeDriver: true,
    };
```

Android don't like Animated.spring in combination with the shared element (even though I copied this from your examples). I changed it to `Animated.timing` on android. It's not actually what I really wanted, but its ok as a workaround.

The next issue was that transitioning the image on the next navigator screen using 
```js
    image: {
        width: "100%",
        height: "100%"
    },
```

was the issue on android for the "stretch" bug. This is working perfectly on iOS though.

I changed it that way:

```js
    image: {
        width: Platform.OS === "android" ? Dimensions.get('window').width : "100%",
        height: Platform.OS === "android" ? Dimensions.get('window').width : "100%"
    },
```

Now I have a pretty consistent view on both android and iOS but iOS don't animate the border, while this is working perfectly on Android.

Is there a workaround or do I really have to wait for EXPO SDK 37?

Thanks for this awesome lib btw.

**Comment by IjzerenHein** [comment]
Hi! The original Android problem that you mentioned is certainly a strange one. It almost seems like that due to the settings, that native-driver is not used. I'm glad you were able to find a solution for that though.

As for the fixes to Android and iOS. There have indeed been several fixes addressing issues with border-radius, borders, FastImage, ImageBackground with the newer RN60+ releases. Some fundamental things were changed on how images & borders were rendered and that broken the image-resolve in certain cases. Android already received some love in that area and I only just fixed the issues on iOS (v0.5.4). Neither of the Android nor iOS fixes have landed in Expo SDK 36 though. I'll make sure they land in Expo 37 though!

**Comment by Hirbod** [comment]
Thanks. If you have time I would love if you could check that spring issue (maybe its already fixed in 0.5.4, but I can't test that unfortunately.

**Comment by IjzerenHein** [comment]
Well the best way forward would to create a test and showcases the problem. If you could help create the test then I can take a closer look at it on Android. The test can be added to the `./Example` app in the repo. You might need to extend the test-framework so that you can pass in a custom transition/timing function on a per test basis though.
